### PR TITLE
TransactionBuilder should include computeBudgetLimit for fixed budget option

### PR DESCRIPTION
**Pull request by zach225**
Also includes logic to only include the compute budget instruction if there is a priority fee

### CODE REVIEW

File: packages/common-sdk/src/web3/transactions/transactions-builder.ts [hunk 1]
```diff
@@ -255,14 +255,16 @@ export class TransactionBuilder {
         (computeBudgetOption.priorityFeeLamports * MICROLAMPORTS_PER_LAMPORT) / computeLimit,
       );
 
-      prependInstructions = [
```
Comment (zach225) [inline packages/common-sdk/src/web3/transactions/transactions-builder.ts:258]: I'm guessing both should be inside the if statement, but need to double check

Comment (wjthieme) [reply packages/common-sdk/src/web3/transactions/transactions-builder.ts:258]: I think we need to always include computeUnitLimit but can optionally exclude computeLimitPrice if it is 0

File: packages/common-sdk/src/web3/transactions/transactions-builder.ts [hunk 2]
```diff
@@ -397,9 +399,13 @@ export class TransactionBuilder {
...
     let finalComputeBudgetOption = computeBudgetOption ?? { type: "none" };
+    let margin = 0.1;
+
     if (finalComputeBudgetOption.type === "auto") {
-      const margin = finalComputeBudgetOption.computeLimitMargin ?? 0.1;
-      const lookupTableAccounts =
+      margin = finalComputeBudgetOption.computeLimitMargin ?? margin;
+    }
+
+    const lookupTableAccounts =
```
Comment (wjthieme) [inline packages/common-sdk/src/web3/transactions/transactions-builder.ts:408]: This estimates computeBudgetLimit even if a static number is set in the `budgetOption`. Can we skip the estimation if a static number is set?

File: packages/common-sdk/src/web3/transactions/transactions-builder.ts (hunk 1 above)
Comment (zach225) [reply packages/common-sdk/src/web3/transactions/transactions-builder.ts]: thanks for the comment. `fixed` now only includes `computeUnitPrice` if `microLamports` is greater than 0

File: packages/common-sdk/src/web3/transactions/transactions-builder.ts (hunk 2 above)
Comment (zach225) [reply packages/common-sdk/src/web3/transactions/transactions-builder.ts:403]: updated so we only estimate the limit if `computeBudgetLimit` is not initially passed in for `fixed` type. checking for `undefined` to handle scenario where `0` is passed as value

**Review by wjthieme — APPROVED** [review]

**Pull request merged**